roles/openshift_named_certificates/tasks/main.yml

```diff
@@ -40,7 +47,29 @@
 
 - name: Land named CA certificates
   copy:
-    src: "{{ item }}"
-    dest: "{{ named_certs_dir }}/{{ item | basename }}"
+    src: "{{ item.cafile }}"
+    dest: "{{ named_certs_dir }}/{{ item.cafile | basename }}"
+    mode: 0600
+  with_items: "{{ named_certificates }}"
+  when: "'cafile' in item"
+
+- name: Land named CA certificates in temporary fragment directories
+  copy:
+    src: "{{ item.cafile }}"
+    dest: "{{ named_certs_dir }}/fragments/{{ item.certfile | basename }}/01_{{ item.cafile | basename }}"
+    mode: 0600
+  with_items: "{{ named_certificates }}"
+  when: "'cafile' in item"
+
+- name: Compile cert files from fragments
+  assemble:
+    src: "{{ named_certs_dir }}/fragments/{{ item.certfile | basename }}"
+    dest: "{{ named_certs_dir }}/{{ item.certfile | basename }}"
     mode: 0600
-  with_items: "{{ named_certificates | oo_collect('cafile') }}"
```

Comment:
I assume oo_collect was used in place of the when statements that I implemented. I'm not sure why it would be preferred.